File: android/app/src/main/java/com/noctis/app/model/Languages.kt
package com.noctis.app.model

data class LanguageOption(val code: String, val label: String)

/** Mirrors web/src/lib/languages.ts — drives the picker only, doesn't limit what ML Kit can translate. */
val SUPPORTED_LANGUAGES = listOf(
    LanguageOption("pt", "Português"),
    LanguageOption("en", "English"),
    LanguageOption("es", "Español"),
    LanguageOption("fr", "Français"),
    LanguageOption("de", "Deutsch"),
    LanguageOption("it", "Italiano"),
    LanguageOption("ja", "Japonês (日本語)"),
    LanguageOption("ko", "Coreano (한국어)"),
    LanguageOption("zh-CN", "Chinês simplificado (中文)"),
    LanguageOption("zh-TW", "Chinês tradicional (中文)"),
    LanguageOption("ru", "Russo (Русский)"),
    LanguageOption("ar", "Árabe (العربية)"),
    LanguageOption("hi", "Hindi (हिन्दी)"),
    LanguageOption("nl", "Neerlandês (Nederlands)"),
    LanguageOption("pl", "Polonês (Polski)"),
    LanguageOption("tr", "Turco (Türkçe)"),
    LanguageOption("vi", "Vietnamita (Tiếng Việt)"),
    LanguageOption("id", "Indonésio (Bahasa Indonesia)"),
)

File: web/src/lib/languages.ts
/**
 * Curated language list for the UI. This does NOT limit which languages can
 * actually be translated — any provider that supports a code not listed here
 * still works if passed directly (e.g. from `?lang=` in the URL). The list
 * only drives the language picker dropdown, so it exists to keep that menu
 * usable rather than to gate functionality.
 */
export interface LanguageOption {
  code: string;
  label: string;
}

export const LANGUAGES: LanguageOption[] = [
  { code: 'pt', label: 'Português' },
  { code: 'en', label: 'English' },
  { code: 'es', label: 'Español' },
  { code: 'fr', label: 'Français' },
  { code: 'de', label: 'Deutsch' },
  { code: 'it', label: 'Italiano' },
  { code: 'ja', label: 'Japonês (日本語)' },
  { code: 'ko', label: 'Coreano (한국어)' },
  { code: 'zh-CN', label: 'Chinês simplificado (中文)' },
  { code: 'zh-TW', label: 'Chinês tradicional (中文)' },
  { code: 'ru', label: 'Russo (Русский)' },
  { code: 'ar', label: 'Árabe (العربية)' },
  { code: 'hi', label: 'Hindi (हिन्दी)' },
  { code: 'nl', label: 'Neerlandês (Nederlands)' },
  { code: 'pl', label: 'Polonês (Polski)' },
  { code: 'tr', label: 'Turco (Türkçe)' },
  { code: 'vi', label: 'Vietnamita (Tiếng Việt)' },
  { code: 'th', label: 'Tailandês (ไทย)' },
  { code: 'id', label: 'Indonésio (Bahasa Indonesia)' },
  { code: 'sv', label: 'Sueco (Svenska)' },
];

export const DEFAULT_TARGET_LANG = 'pt';

export function isKnownLanguage(code: string): boolean {
  return LANGUAGES.some((l) => l.code === code);
}
